src/tests/conftest.py

```diff
@@ -0,0 +1,103 @@
+import datetime
+import os.path
+from collections import namedtuple
+from contextlib import closing
+from pathlib import Path
+
+import pytest
+from airflow import DAG
+from airflow.models import Connection
+from psycopg2 import extras
+from pytest_docker_tools import container, fetch
+
+pytest_plugins = ["helpers_namespace"]
+
+
+@pytest.fixture
+def test_dag():
+    return DAG(
+        "test_dag",
+        default_args={"owner": "airflow", "start_date": datetime.datetime(2018, 1, 1)},
+        schedule_interval=datetime.timedelta(days=1),
+    )
+
+
+@pytest.helpers.register  # type: ignore
+def run_task(task, dag):
+    dag.clear()
+    task.run(
+        start_date=dag.default_args["start_date"],
+        end_date=dag.default_args["start_date"],
+    )
+
+
+@pytest.fixture(scope="module")
+def postgres_credentials():
+    PostgresCredentials = namedtuple("PostgresCredentials", ["username", "password"])
+    return PostgresCredentials("testuser", "testpass")
+
+
+postgres_image = fetch(repository="postgres:11.1-alpine")  # type: ignore
+
+
+postgres = container(  # type: ignore
+    image="{postgres_image.id}",
+    environment={
+        "POSTGRES_USER": "{postgres_credentials.username}",
+        "POSTGRES_PASSWORD": "{postgres_credentials.password}",
+    },
+    ports={"5432/tcp": None},
+)
+
+
+@pytest.fixture
+def mock_pghook(mocker, postgres, postgres_credentials):
+    def _apply_mock(PostgresHook):
+        mocker.patch.object(
+            PostgresHook,
+            "get_connection",
+            return_value=Connection(
+                host="localhost",
+                conn_type="postgres",
+                login=postgres_credentials.username,
+                password=postgres_credentials.password,
+                port=postgres.ports["5432/tcp"][0],
+            ),
+        )
+
+    return _apply_mock
+
+
+@pytest.fixture
+def pg_cursor(mocker, postgres, postgres_credentials, request):
+
+    # Use the request.module to get PostgresHook
+    # from the requesting module. By keeping this PostgresHook
+    # dynamic, this fixture can be use to test different airflow modules.
+    mocker.patch.object(
+        request.module.PostgresHook,
+        "get_connection",
+        return_value=Connection(
+            host="localhost",
+            conn_type="postgres",
+            login=postgres_credentials.username,
+            password=postgres_credentials.password,
+            port=postgres.ports["5432/tcp"][0],
+        ),
+    )
+    hook = request.module.PostgresHook()
+    conn = hook.get_conn()
```

Comment:
I have read the comment that `close()` might not be necessary. However as you still close the `conn` explicitly you might as well use: `with closing(hook.get_conn()) as conn:` That way the `conn.close()` will be called in the presence of exceptions.

Reply:
Yeah. I tried several approaches and thought that yielding from the `closing` context mgr. from a fixture does not work. But, it does. So, I'll change it and also get the cursor from a context mgr.
Initially I did not realize that the container fixture has function scope, so is postgres is re-initialized anyway.

Reply:
Yeah, context managers and yield work fine together. It's almost magic.

Reply:
Yeah, same goes for the connection, I could remove `closing()` for both connection and cursor.